raiden/network/transport/matrix.py

```diff
@@ -188,26 +189,78 @@ def _http_retry_delay() -> Iterable[float]:
     def start(
         self,
         raiden_service: RaidenService,
-        queueids_to_queues: Dict[QueueIdentifier, List[Event]],
+        initial_queues: Dict[QueueIdentifier, List[SendMessageEvent]],
     ):
-        self._running = True
+        if not self._stop_event.ready():
+            return  # already started
+        self._stop_event.clear()
         self._raiden_service = raiden_service
 
         self._login_or_register()
         self._join_discovery_room()
         self._inventory_rooms()
 
-        # TODO: Add (better) error handling strategy
         self._client.start_listener_thread()
+        self._client.sync_thread.link_exception(self.on_error)
+        # no need to link_value on long-lived sync_thread, as it shouldn't succeed but on stop
+        # cleint's sync thread may also crash self/main greenlet
+        self.greenlets = [self._client.sync_thread]
 
-        # TODO: Add greenlet that regularly refreshes our presence state
         self._client.set_presence_state(UserPresence.ONLINE.value)
+        self._send_queued_messages()  # uses property instead of initial_queues
 
-        gevent.spawn_later(1, self._send_queued_messages, queueids_to_queues)
         self.log.info('TRANSPORT STARTED')
 
+        super().start()  # start greenlet
+
+    def run(self):
+        """ Runnable main method, perform wait on long-running subtasks """
+        try:
+            # children crashes should throw an exception here
+            self._stop_event.wait()
+        finally:
+            self.stop()  # noop on double call, ensure cleanup and wait on subtasks
+
+    def stop(self, timeout=None):
+        """ Try to gracefully stop the greenlet synchronously
+
+        Stop isn't expected to re-raise greenlet _run exception
+        (use self.greenlet.get() for that),
+        but it should raise any stop-time exception """
+        if self._stop_event.ready():
+            return
+        self._stop_event.set()
+        if timeout is None:
+            timeout = self._config.get('logout_timeout', 10)
+
+        self._client.set_presence_state(UserPresence.OFFLINE.value)
+
+        try:
+            with gevent.Timeout(timeout, 'Logging out despite unjoined greenlets.'):
```

Comment:
we already have a timeout on the raiden service, no need for this here

Reply:
All `Runnable` have the `timeout` parameter for `stop`, I'm just sticking to the interface/behavior, it was already there, just updated for the new interface.

Reply:
Please don't add the timeout here, we have had multiple discussions about this and timeouts don't compose.